src/controllers/Auth.controller.js

```diff
@@ -0,0 +1,76 @@
+const bcrypt = require('bcryptjs');
+const jwt = require('jsonwebtoken');
+require('env2')('config.env');
+
+const { SECRET } = process.env;
+const StudentController = require('./Student.controller');
+
+
+const hashStudentPassword = student => new Promise((resolve, reject) => {
+  bcrypt.hash(student.password, 10, (hashError, hashedPassword) => {
+    if (hashError) {
+      reject(hashError);
+    } else {
+      const newStudent = student;
+      newStudent.password = hashedPassword;
+      resolve(newStudent);
+    }
+  });
+});
+
+const generateIdCookie = ({ id }) => new Promise((resolve, reject) => {
+  jwt.sign({ id }, SECRET, (signErr, token) => {
+    if (signErr || !token) {
+      reject(signErr || new Error('No token has generated!'));
+    } else {
+      resolve(token);
+    }
+  });
+});
+
+const matchPasswords = (student, password) => new Promise((resolve, reject) => {
+  bcrypt.compare(password, student.password)
+    .then((matchingState) => {
+      if (!matchingState) {
+        reject(new Error('Password is not correct!'));
+      }
+      resolve(student);
+    })
+    .catch((compareErr) => {
+      reject(compareErr);
+    });
+});
+
+const signup = (req, res) => {
+  const student = req.body;
+  if (!student) {
+    res.json({ success: false, err: 'No student data found!' });
```

Comment:
Consdier the scinario where the `req.body` is empty, the user haven't filled any field, `req.body = {}` and hence `{}` always evalutes as `true`. 
Also the students should have multiple varibale such as gender, age..etc, so if you are trying to do server-side validation you should check for each variable